tdrs-backend/tdpservice/users/api/authorization_check.py

```diff
@@ -14,6 +15,7 @@ class AuthorizationCheck(APIView):
 
     query_string = False
     pattern_name = "authorization-check"
+    permission_classes = [IsAuthenticatedOrReadOnly]
```

Comment:
This was to allow the system to respond to the authorization check with a proper message rather than just a 403

Reply:
Is this change directly related to the criteria in #231, or is it an incidental change? 

Also, would you mind saying a little more about why you are using `IsAuthenticatedOrReadOnly` here? 

The Django REST Framework docs say:

> The IsAuthenticatedOrReadOnly will allow authenticated users to perform any request. Requests for unauthorised users will only be permitted if the request method is one of the "safe" methods; GET, HEAD or OPTIONS.
> 
> This permission is suitable if you want to your API to allow read permissions to anonymous users, and only allow write permissions to authenticated users.

https://www.django-rest-framework.org/api-guide/permissions/

Reply:
@alexsoble 
If you look at the backend tasks, you will see an item `Create tests to cover lines 22->37, and 37->41 of the authorization check endpoint to complete test coverage`

This was added as an incidental task, not really related to the AC, but since we are discouraging creating separate tickets for things like this, Laura said we should just add it as a task to an upcoming issue, and this is where it landed. This particular bit was just part of that cleanup.

As for the particular permission, I think you are right - this should be AllowAny - I will update it.

Reply:
It's updated now

Reply:
Gotcha, can you say more about why you think AllowAny is the right permissions to set here?

Reply:
We needed to set it because the system is set to only allow authenticated users as an overall app setting. Since this check is to explicitly return whether a user is authenticated or not, the endpoint needs to allow all requests to return properly. Without it it returns a 403.

> The AllowAny permission class will allow unrestricted access, regardless of if the request was authenticated or unauthenticated.

https://www.django-rest-framework.org/api-guide/permissions/